server/server.go

```diff
@@ -550,9 +551,12 @@ func (s *Server) Run() error {
 		}
 	}()
 
-	// Setup gRPC server
-	// for HTTP over gRPC, ensure we don't double-count the middleware
-	httpgrpc.RegisterHTTPServer(s.GRPC, httpgrpc_server.NewServer(s.HTTP))
+	var serverOptions []httpgrpc_server.Option
+	if s.cfg.ReportHTTP4XXCodesInInstrumentationLabel {
+		serverOptions = []httpgrpc_server.Option{httpgrpc_server.WithReturn4XXErrors}
```

Comment:
[nit] Doesn't change the current logic, but makes the intention more clear (you don't want to replace the options, but add an option).

```suggestion
		serverOptions = append(serverOptions, httpgrpc_server.WithReturn4XXErrors)
```